rubicon_ml/viz/experiments_table.py

```diff
@@ -0,0 +1,281 @@
+import dash_bootstrap_components as dbc
+from dash import dash_table, dcc, html
+from dash.dependencies import ALL, Input, Output, State
+
+from rubicon_ml.viz.base import VizBase
+from rubicon_ml.viz.colors import light_blue, plot_background_blue
+
+
+class ExperimentsTable(VizBase):
+    def __init__(self, experiments, is_selectable=True, dash_kwargs={}):
+        super().__init__(dash_kwargs=dash_kwargs, dash_title="rubicon-ml: experiment table")
+
+        self.experiments = experiments
+        self.is_selectable = is_selectable
+
+        self.commit_hash = None
+        self.experiment_records = []
+        self.github_url = None
+        self.metric_names = set()
+        self.parameter_names = set()
+        self.all_columns = ["id", "name", "created_at", "model_name", "commit_hash", "tags"]
+        self.hidden_columns = []
+
+        commit_hashes = set()
+        show_columns = {"id", "created_at"}
+
+        for experiment in self.experiments:
+            if experiment.commit_hash is not None:
+                commit_hashes.add(experiment.commit_hash)
+                show_columns.add("commit_hash")
+
+            if experiment.model_name is not None:
+                show_columns.add("model_name")
+
+            if experiment.name is not None:
+                show_columns.add("name")
+
+            if len(experiment.tags) > 0:
+                show_columns.append("tags")
+
+            experiment_record = {
+                "id": experiment.id,
+                "name": experiment.name,
+                "created_at": experiment.created_at,
+                "model_name": experiment.model_name,
+                "commit_hash": experiment.commit_hash[:7],
+                "tags": experiment.tags,
+            }
+
+            for parameter in experiment.parameters():
+                experiment_record[parameter.name] = parameter.value
+
+                self.parameter_names.add(parameter.name)
+
+            for metric in experiment.metrics():
+                experiment_record[metric.name] = metric.value
+
+                self.metric_names.add(metric.name)
+
+            self.experiment_records.append(experiment_record)
+
+        self.metric_names = list(self.metric_names)
+        self.parameter_names = list(self.parameter_names)
+
+        self.all_columns.extend(self.parameter_names + self.metric_names)
+        self.hidden_columns = [
+            column
+            for column in self.all_columns
+            if column not in list(show_columns) + self.metric_names + self.parameter_names
+        ]
+
+        if len(commit_hashes) == 1:
+            github_url_root = self.experiments[0].project.github_url[:-4]
+            commit_hash = list(commit_hashes)[0]
+
+            self.commit_hash = commit_hash[:7]
+            self.github_url = f"{github_url_root}/tree/{commit_hash}"
+
+        self.app.layout = self._build_frame(self._build_layout())
+
+        _register_callbacks(self.app)
+
+    def _build_layout(self):
```

Comment:
What are the intended functions of these buttons? When I run this in the notebook, it doesn't look like there are any changes being made to the visualization.

Reply:
the select and clear all experiment buttons right? they're intentionally disabled in the context of a single experiment table because selecting experiment rows doesn't have any value in the standalone experiment table

when the experiment table is rendered in the dashboard with another widget(s) that takes experiments as an input, the experiment table rows will be selectable and selecting them will update the inputs to the other widget(s)

it may be better to make the buttons disappear in the standalone widget, rather than disable them. or maybe I'll just add some kind of tooltip to explain why they're disabled

Reply:
@andreafehrman good call - I ended up hiding them instead of just disabling them and I think it looks a lot better and less confusing